src/analysis/processing/qgsalgorithmrasterminmax.cpp

```diff
@@ -0,0 +1,237 @@
+                      "If multiple pixels in the raster share the minimum or maximum value, then only one of these pixels will be included in the output." );
+}
+
+QString QgsRasterMinMaxAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Calculates the minimum and maximum pixel in a raster layer." );
+}
+
+QgsRasterMinMaxAlgorithm *QgsRasterMinMaxAlgorithm::createInstance() const
+{
+  return new QgsRasterMinMaxAlgorithm();
+}
+
+bool QgsRasterMinMaxAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *layer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT" ), context );
+  const int band = parameterAsInt( parameters, QStringLiteral( "BAND" ), context );
+
+  if ( !layer )
+    throw QgsProcessingException( invalidRasterError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  mBand = parameterAsInt( parameters, QStringLiteral( "BAND" ), context );
+  if ( mBand < 1 || mBand > layer->bandCount() )
+    throw QgsProcessingException( QObject::tr( "Invalid band number for BAND (%1): Valid values for input raster are 1 to %2" ).arg( mBand )
+                                  .arg( layer->bandCount() ) );
+
+  mInterface.reset( layer->dataProvider()->clone() );
+  mHasNoDataValue = layer->dataProvider()->sourceHasNoDataValue( band );
+  mLayerWidth = layer->width();
+  mLayerHeight = layer->height();
+  mExtent = layer->extent();
+  mCrs = layer->crs();
+  mRasterUnitsPerPixelX = std::abs( layer->rasterUnitsPerPixelX() );
+  mRasterUnitsPerPixelY = std::abs( layer->rasterUnitsPerPixelY() );
+  return true;
+}
+
+QVariantMap QgsRasterMinMaxAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  QString dest;
+  std::unique_ptr< QgsFeatureSink > sink;
+  if ( parameters.value( QStringLiteral( "OUTPUT" ) ).isValid() )
+  {
+    QgsFields outFields;
+    outFields.append( QgsField( QStringLiteral( "value" ), QMetaType::Type::Double, QString(), 20, 8 ) );
+    outFields.append( QgsField( QStringLiteral( "extremum" ), QMetaType::Type::QString ) );
+    sink.reset( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, dest, outFields, Qgis::WkbType::Point, mCrs ) );
+    if ( !sink )
+      throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+  }
+
+  const int extractType = parameterAsInt( parameters, QStringLiteral( "EXTRACT" ), context );
+
+  const int maxWidth = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_WIDTH;
+  const int maxHeight = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_HEIGHT;
+  const int numberBlocksWidth = static_cast< int >( std::ceil( 1.0 * mLayerWidth / maxWidth ) );
+  const int numberBlocksHeight = static_cast< int >( std::ceil( 1.0 * mLayerHeight / maxHeight ) );
+  const int numberBlocks = numberBlocksWidth * numberBlocksHeight;
```

Comment:
highly unlikely to overflow, but who knows...
```suggestion
  const qint64 numberBlocks = static_cast<qint64>(numberBlocksWidth) * numberBlocksHeight;
```

Reply:
I wonder if a raster that big exists anywhere :laughing:

Reply:
> I wonder if a raster that big exists anywhere 😆

a real one, perhaps not, but you could easily create an hostile very small VRT with huge dimensions... (I've fixed so many integer overflows while fixing hundreds of ossfuzz GDAL bugs that pretty much any integer addition and multiplication has become suspicious to my eyes)

Reply:
Never challenge me for that kind of game ;-) Here's such a GeoPackage: [huge.gpkg.zip](https://github.com/user-attachments/files/17625676/huge.gpkg.zip) : 100 million pixel x 100 million pixel in 98 KB